pkg/util/ecs/metadata/clients.go

```diff
@@ -179,3 +206,12 @@ func initV3orV4() error {
 	globalUtil.v3or4 = client
 	return nil
 }
+
+func initV4() error {
+	client, err := newClientV4ForCurrentTask()
+	if err == nil {
+		globalUtil.v4 = client
+		return nil
+	}
+	return err
+}
```

Comment:
The `if` may not be necessary.

```suggestion
func initV4() (err error) {
	globalUtil.v4, err = newClientV4ForCurrentTask()
	return err
}
```